src/main/scala/cromwell/Main.scala

```diff
@@ -227,11 +247,52 @@ object Main extends App {
         |  Starts a web server on port 8000.  See the web server
         |  documentation for more details about the API endpoints.
       """.stripMargin)
-    if (exit) System.exit(-1)
+    exit(-1)
+  }
+
+  private[this] def initLogging(): Int = {
+    val systemProperties = sys.props
+    val logRoot = systemProperties.getOrElseUpdate("LOG_ROOT", File("").fullPath)
+    systemProperties.getOrElseUpdate("LOG_MODE", "CONSOLE")
+    systemProperties.getOrElseUpdate("LOG_LEVEL", "INFO")
+
+    try {
+      File(logRoot).createDirectories()
+      0
+    } catch {
+      case e: Throwable =>
+        Console.err.println(s"Could not create log directory: $logRoot")
+        e.printStackTrace()
+        exit(1)
+    }
   }
 
-  def getAction(firstArg: Option[String]): Option[Actions.Value] = for {
+  private[this] def continueIf(valid: => Boolean)(block: => Int): Int = if (valid) block else usageAndExit()
+
+  private[this] def getAction(firstArg: Option[String]): Option[Actions.Value] = for {
     arg <- firstArg
-    a <- Actions.values find { _.toString == arg }
+    argCapitalized = arg.capitalize
+    a <- Actions.values find (_.toString == argCapitalized)
   } yield a
+
+  private[this] def loadWdl(path: String)(f: WdlNamespace => Int): Int = {
+    Try(WdlNamespace.load(new JFile(path), WorkflowManagerActor.BackendType)) match {
+      case Success(namespace) => f(namespace)
+      case Failure(t) =>
+        println(t.getMessage)
+        exit(1)
+    }
+  }
+
+  // shortcut
+  private[this] def loadWdl(args: Seq[String])(f: WdlNamespace => Int): Int = loadWdl(args.head)(f)
+
+  private[this] def exit(returnCode: Int): Int = {
+    if (enableSysExit) {
+      // $COVERAGE-OFF$Exit not allowed during tests
```

Comment:
or at least not to ding us for uncovered JES code that we know Travis can't reach